htdocs/core/lib/functions.lib.php

```diff
@@ -7145,7 +7145,7 @@ function getCommonSubstitutionArray($outputlangs, $onlykey = 0, $exclude = null,
 				if ($object->fetch_optionals() > 0) {
 					if (is_array($extrafields->attributes[$object->table_element]['label']) && count($extrafields->attributes[$object->table_element]['label']) > 0) {
 						foreach ($extrafields->attributes[$object->table_element]['label'] as $key => $label) {
-							$substitutionarray['__EXTRAFIELD_'.strtoupper($key).'__'] = $object->array_options['options_'.$key];
+							$substitutionarray['__EXTRAFIELD_'.strtoupper($key).'__'] = $extrafields->showOutputField($key, $object->array_options['options_'.$key], '', $object->table_element);
```

Comment:
The showOutputField is a function dedicated to format a date to be output into a HTML content.
But $substitutionarray['__EXTRAFIELD_'.strtoupper($key).'__'] is a variable that is dedicated to contains a brute and not formated value. So we should not apply such a method on the data we want to store into the memory array.

Reply:
Now extrafields can receive many different data: checkbox, picklist, dates, price, boolean, etc...
More and more customers like to use extrafields associated with a substitution variable to create an email and/or a PDF and use these substitution variables even in a product description.
With this PR we avoid duplicating control over the data in order to make it readable and usable.